annotation/app/annotations/main.py

```diff
@@ -358,8 +360,18 @@ def construct_annotated_doc(
     ):
         return latest_doc
 
+    document_links = construct_document_links(
+        annotated_doc, doc.similar_revisions or []
+    )
     db.add(annotated_doc)
-    db.flush()
+    db.add_all(document_links)
+    try:
+        db.flush()  # TODO: shouldn't we commit here?
```

Comment:
if we do db.commit() later, no need to flush here. flush just moves data from python transaction buffer to DB transaction buffer

Reply:
Brief check didn't give the answer where do we commit this transaction =)
Changed to commit